### ISSUE TITLE
Support for different rounding strategies in #number.format

### ISSUE THREAD

**Issue by ghost** [thymeleaf/thymeleaf#581]
Thymleaf 3.0.3 (Spring Boot)

When the decimal of an even number is .5,  the number is rounded down, but is rounded up when the number is odd.

For example: 
`<span th:text="${#numbers.formatInteger(98.5, 0,'COMMA')}"></span>` result=98
`<span th:text="${#numbers.formatInteger(99.5, 0,'COMMA')}"></span>` result=100

The same results occur with `#numbers.formatDecimal` and also when using inline syntax.

**Comment by kazuki43zoo** [comment]
Probably, it is default behavior of `DecimalFormat` provided by JDK.
The [JavaDoc of JDK](https://docs.oracle.com/javase/8/docs/api/java/text/DecimalFormat.html) says as follow:

---

Rounding

`DecimalFormat` provides rounding modes defined in `RoundingMode` for formatting. By default, it uses `RoundingMode.HALF_EVEN`.

---

About `RoundingMode`, see https://docs.oracle.com/javase/8/docs/api/java/math/RoundingMode.html.

Can we customize this behavior?

**Comment by danielfernandez** [comment]
As @kazuki43zoo says, this is due to the default `RoundingMode` being applied by `DecimalFormat`, which is `HALF_EVEN`(when the part of the number to be rounded is equidistant to both extremes, round towards the nearest even number).

There is currently no way to customize the rounding mode to be used in the `#numbers` expression utility objects, though of course different rounding modes can be applied separately, be it directly at the controller, at the objects themselves being rendered, or even by putting a `DecimalFormat` object into the context and calling it directly from the templates.

**Comment by ghost** [comment]
Yeah, that kind of defeats the convenience of using #number.  To reiterate @kazuki43zoo's question, any plans to customize this behavior?

**Comment by topn0tch** [comment]
I'm having the same issue.

**Comment by wiecia** [comment]
Yes, struggling with the same issue. It would be nice to have option to customize it

**Comment by ygogolitsyn** [comment]
Hi, All. My name is Yuri, I received this issue for triage from the CodeTriage site. 

On a positive note – it is possible to add a rounding mode to the utility methods for Numbers, and it is a relatively straightforward task. To avoid backwards compatibility issues a set of new methods with an additional rounding mode parameter should probably be introduced. Due to a fact that a need to change the rounding mode probably emerges in a relatively rare number of scenarios, and to minimise the additional code required the following new methods could be added in the Numbers.java class (CEILING is used as an example of a rounding mode):

${#numbers.formatInteger(num,3,'POINT',’CEILING’)}
${#numbers.arrayFormatInteger(numArray,3,'POINT',’CEILING’)}
${#numbers.listFormatInteger(numList,3,'POINT',’CEILING’)}
${#numbers.setFormatInteger(numSet,3,'POINT',’CEILING’)}	

${#numbers.formatDecimal(num,3,'POINT',2,'COMMA',’CEILING’)}
${#numbers.arrayFormatDecimal(numArray,3,'POINT',2,'COMMA',’CEILING’)}
${#numbers.listFormatDecimal(numList,3,'POINT',2,'COMMA',’CEILING’)}
${#numbers.setFormatDecimal(numSet,3,'POINT',2,'COMMA',’CEILING’)}

It would be required to make modifications in the NumberUtils.java class – add a “format” method with the additional rounding parameter, add (or modify) the “formatNumber” method that internally instantiates the DecimalFormat object and provide a convenience method to handle the rounding mode strings.

On the negative side – as it is mentioned in the comment by @danielfernandez of 26th Mar 2017, different rounding modes can be applied separately using several options. In other words, not being able to specify the rounding mode in #number is not a showstopper, but an inconvenience.

it needs to be weighted and agreed whether the convenience of using #number to specify the rounding mode justifies adding the above additional code.

**Comment by jorgevl** [comment]
This would come incredibly handy, 100%.
